tests/utils/test_metrics.py

```diff
@@ -0,0 +1,170 @@
+    pred_multi = torch.tensor([0, 0, 2, 2, 0, 2, 1, 2, 1, 0, 2, 2, 1, 0, 2, 2])
+    pred_binary = torch.tensor([0, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1])
+
+    lbl_multi = torch.tensor([1, 0, 2, 2, 0, 1, 2, 0, 2, 2, 0, 0, 1, 2, 0, 1])
+    lbl_binary = torch.tensor([1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1])
+    # array with dont care
+    lbl_multi_dc = torch.tensor([-1, -1, 2, 2, 0, 1, 2, 0, 2, 2, 0, 0, 1, 2, 0, 1])
+    lbl_binary_dc = torch.tensor([-1, -1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1])
+    return {'pred_multi': pred_multi, 
+            'pred_binary': pred_binary, 
+            'lbl_multi': lbl_multi, 
+            'lbl_binary': lbl_binary, 
+            'lbl_multi_dc': lbl_multi_dc, 
+            'lbl_binary_dc': lbl_binary_dc}
+
+
+class TestMetrics(object):
+    def test_create_metrics_dict(self):
+        """Evaluate the metrics dictionnary creation. 
+        Binary and multiclass"""
+        # binary tasks have 1 class at class definition.
+        num_classes = 1
+        metrics_dict = create_metrics_dict(num_classes)
+        assert 'iou_1' in metrics_dict.keys()
+        assert 'iou_2' not in metrics_dict.keys()
+
+        num_classes = 3
+        metrics_dict = create_metrics_dict(num_classes)
+        assert 'iou_1' in metrics_dict.keys()
+        assert 'iou_2' in metrics_dict.keys()
+        assert 'iou_3' not in metrics_dict.keys()
+        del metrics_dict
+
+    def test_report_classification_multi(self):
+        """Evaluate report classification. 
+        Multiclass, without ignore_index in array."""
+        t = init_tensors()
+        metrics_dict = create_metrics_dict(3)
+        metrics_dict = report_classification(t['pred_multi'], 
+                                             t['lbl_multi'], 
+                                             batch_size=2, 
+                                             metrics_dict=metrics_dict,
+                                             ignore_index=-1)
+        assert "{:.6f}".format(metrics_dict['precision'].val) == "0.327083"
+        assert "{:.6f}".format(metrics_dict['recall'].val) == "0.312500"
+        assert "{:.6f}".format(metrics_dict['fscore'].val) == "0.314935"
+
+    def test_report_classification_multi_ignore_idx(self):
+        """Evaluate report classification. 
+        Multiclass, with ignore_index in array."""
+        t = init_tensors()
+        metrics_dict = create_metrics_dict(3)
+        metrics_dict = report_classification(t['pred_multi'], 
+                                             t['lbl_multi_dc'], 
+                                             batch_size=2, 
+                                             metrics_dict=metrics_dict,
+                                             ignore_index=-1)
+        assert "{:.6f}".format(metrics_dict['precision'].val) == "0.297619"
+        assert "{:.6f}".format(metrics_dict['recall'].val) == "0.285714"
+        assert "{:.6f}".format(metrics_dict['fscore'].val) == "0.283163"
```

Comment:
If you'd like to one step further in this verification, you could define some functions for precision, recall, fscore and iou from numpy directly and compare results from torchmetrics with these simple functions. This assertion is enough in my opinion though. The only downside is that it doesn't "prove" that a metric is being calculated the right way, it just proves that it's constant.

Reply:
Right. I took the easy path of calculating the values in an excel sheet, to validate them.

Reply:
That's perfect. Thanks for validating!